torch/csrc/distributed/rpc/request_callback_impl.cpp

```diff
@@ -224,15 +229,18 @@ void RequestCallbackImpl::processRpc(
 
       auto& pythonRpcHandler = PythonRpcHandler::getInstance();
       IValue py_ivalue;
-      {
-        pybind11::gil_scoped_acquire ag;
-        py_ivalue = jit::toIValue(
-            pythonRpcHandler.runPythonUdf(std::move(uprc).movePythonUdf()),
-            PyObjectType::get());
+      try {
+        {
+          pybind11::gil_scoped_acquire ag;
+          py_ivalue = jit::toIValue(
+              pythonRpcHandler.runPythonUdf(std::move(uprc).movePythonUdf()),
+              PyObjectType::get());
+        }
+        ownerRRef->setValue(std::move(py_ivalue));
+      } catch (std::exception& e) {
```

Comment:
For Python UDF, we might get `py::error_already_set` error, which requires GIL to destruct  (see #35283 )

Reply:
Thanks for the pointer! I went ahead and copied the recipe from that PR.